app/src/main/java/com/iic/shopingo/ui/request_flow/activities/RequestStateActivity.java

```diff
@@ -20,27 +23,36 @@
   @Override
   protected void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
-    int state = getIntent().getIntExtra(REQUEST_STATE_EXTRA_KEY, 0);
-    setContentView(LAYOUTS[state]);
+    request = getIntent().getParcelableExtra(REQUEST_EXTRA_KEY);
+    setContentView(LAYOUTS[request.status.ordinal()]);
     ButterKnife.inject(this);
   }
 
   @Optional
   @OnClick(R.id.request_state_cancel_button)
   public void onCancelRequest(View view) {
-    // TODO: Cancel request in server and go to trippers list
+    // TODO: Cancel request in server
+    request.shopper = null;
+    Intent intent = new Intent(this, SelectShopperActivity.class);
+    intent.putExtra(SelectShopperActivity.REQUEST_EXTRA_KEY, request);
+    startActivity(intent);
   }
 
   @Optional
   @OnClick(R.id.request_state_settle_button)
   public void onSettleRequest(View view) {
-    // TODO: Settle request in server and go home
+    // TODO: Settle request in server
+    Intent intent = new Intent(this, HomeActivity.class);
+    startActivity(intent);
```

Comment:
after `startActivity` you need to call `finish`, as you don't want to allow the user back to return him to this activity.